api_core/google/api_core/gapic_v1/routing_header.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2017 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helpers for handling routing header params."""
+
+
+from six.moves.urllib.parse import urlencode
+
+ROUTING_METADATA_KEY = 'x-goog-header-params'
+
+
+def to_routing_header(params):
+    """Returns the routing header string that the params form"""
+    return urlencode(params)
```

Comment:
So it's at this point I feel that I need to ask: what is the benefit of having this module at all if all it does it directly call `urlencode`. If the answer is "we may change how this works under the covers but keep the same interface" then cool.

/cc @lukesneeringer

Reply:
You guessed it. The initial implementation is simple, but it will change.

Reply:
![coolbeans](https://user-images.githubusercontent.com/250995/32468209-335c7b46-c302-11e7-8a37-c7df019418a8.gif)

Since this is a public API, I'm gonna make some documentation suggestions here, then we can merge.